### PR TITLE
#6405 Add annotations wcag

### LINKED ISSUE #6405: annotate abbreviations and SecureDrop jargon (WCAG 3.1.3, 3.1.4)
*This is a good first issue for new contributors to take on, if you have any questions, please ask on the task or in our [Gitter room](https://gitter.im/freedomofpress/securedrop)!*

## Description

[Accessibility Lab recommendations](https://docs.google.com/document/d/1jZlKco2YgOrrI4PmTkG_Q8lKuV5VxCds/edit?disco=AAAAIplLpuQ):

> A mechanism is available for identifying specific definitions of words or phrases used in an unusual or restricted way, including idioms and jargon.
> 
> A mechanism for identifying the expanded form or meaning of abbreviations is available.

## User Research Evidence

[WCAG 3.1.3](https://www.w3.org/WAI/WCAG21/Understanding/unusual-words) and [3.1.4](https://www.w3.org/WAI/WCAG21/Understanding/abbreviations) per #5972.

## Implementation

Suggested as part of #6211:  Is digital-security and SecureDrop-specific jargon marked with `<abbr title>`, ideally drawing from the [SecureDrop glossary](https://weblate.securedrop.org/browse/securedrop/glossary/en/?q=)?

### PR DESCRIPTION
**Pull request by ghost**
## Status

Ready for review / Work in progress

## Description of Changes
Added abbreviation descriptions to the html pages.
Fixes #6405.

Changes proposed in this pull request:
- add abbr tag to html pages
## Testing

How should the reviewer test this PR?
Write out any special testing steps here.

- when looking through the page ensure the correct words have abbreviations 
## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you added or removed a file deployed with the application:

- [ ] I have updated AppArmor rules to include the change

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

Choose one of the following:

- [ x] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ x] I would appreciate help with the documentation
- [ ] These changes do not require documentation

### If you added or updated a reference to a production code dependency:

Production code dependencies are defined in:

- `admin/requirements.in`
- `admin/requirements-ansible.in`
- `securedrop/requirements/python3/requirements.in`
- `securedrop/requirements/python3/translation.in` (used in the build
  container)

If you changed another `requirements.in` file that applies only to development
or testing environments, then no diff review is required, and you can skip
(remove) this section.

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review
- [ ] I am silencing an alert related to a production dependency, because (please explain below):

### CODE REVIEW

File: securedrop/journalist_templates/account_new_two_factor.html [hunk 1]
```diff
@@ -4,15 +4,15 @@
 
 {% block body %}
 {% if user.is_totp %}
-<h1>{{ gettext('Enable FreeOTP') }}</h1>
+<h1>{{ gettext('Enable <abbr title="two-factor authentication application">FreeOTP</abbr>') }}</h1>
```
Comment (cfm) [inline securedrop/journalist_templates/account_new_two_factor.html:7]: "FreeOTP" is the name of a specific application, so it doesn't need a generic annotation.

File: securedrop/journalist_templates/account_new_two_factor.html [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 <p>
   {{ gettext("You're almost done! To finish resetting your two-factor authentication, follow the instructions below to set up FreeOTP. Once you've added the entry for your account in the app, enter one of the 6-digit codes from the app to confirm that two-factor authentication is set up correctly.") }}
 </p>
 
 <ol>
   <li>{{ gettext('Install FreeOTP on your phone') }}</li>
   <li>{{ gettext('Open the FreeOTP app') }}</li>
-  <li>{{ gettext('Tap the QR code symbol at the top') }}</li>
+  <li>{{ gettext('Tap the <abbr title="quick response">QR</abbr> code symbol at the top') }}</li>
```
Comment (cfm) [inline securedrop/journalist_templates/account_new_two_factor.html:15]: I wouldn't annotate an initialism that's (a) in common use and (b) cited far more widely (463 million Google results) than its expansion (2.1 million results).

File: securedrop/journalist_templates/account_new_two_factor.html [hunk 3]
```diff
@@ -23,7 +23,7 @@ <h1>{{ gettext('Enable FreeOTP') }}</h1>
 <mark id="shared-secret">{{ user.formatted_otp_secret }}</mark>
 <p>{{ gettext("Once you have paired FreeOTP with this account, enter the 6-digit verification code below:") }}</p>
 {% else %}
-<h1>{{ gettext('Enable YubiKey (OATH-HOTP)') }}</h1>
+<h1>{{ gettext('Enable YubiKey (<abbr title="HMAC based one-time password algorithm">OATH-HOTP</abbr>)') }}</h1>
```
Comment (cfm) [inline securedrop/journalist_templates/account_new_two_factor.html:26]:
```suggestion
<h1>{{ gettext('Enable YubiKey (OATH-<abbr title="HMAC-based one-time password">HOTP</abbr>)') }}</h1>
```

File: securedrop/journalist_templates/admin.html [hunk 4]
```diff
@@ -85,7 +85,7 @@ <h2 id="users-heading" class="visually-hidden">Users</h2>
 
 <a href="{{ url_for('admin.manage_config') }}" class="btn icon icon-edit section-spacing-inline"
   id="update-instance-config" aria-label="{{ gettext('Update instance config') }}">
```
Comment (cfm) [inline securedrop/journalist_templates/admin.html:87]: See next comment:

```suggestion
  id="update-instance-config" aria-label="{{ gettext('Update instance configuration') }}">
```

File: securedrop/journalist_templates/admin.html [hunk 5]
```diff
@@ -85,7 +85,7 @@ <h2 id="users-heading" class="visually-hidden">Users</h2>
...
-  {{ gettext('INSTANCE CONFIG') }}
+  {{ gettext('INSTANCE <abbr title="Configuration">CONFIG</abbr>') }}
```
Comment (cfm) [inline securedrop/journalist_templates/admin.html:88]: Let's go ahead and just expand this, as we have in:  :-)

https://github.com/freedomofpress/securedrop/blob/4d9b74b7e494b6a8e2d06677bc7c20d558b5f078/securedrop/journalist_templates/config.html#L10

```suggestion
  {{ gettext('INSTANCE CONFIGURATION') }}
```

File: securedrop/journalist_templates/config.html [hunk 6]
```diff
@@ -15,7 +15,7 @@ <h2 id="config-orgname">{{ gettext('Organization Name') }}</h2>
   <form action="{{ url_for('admin.update_org_name') }}" method="post">
     <input name="csrf_token" type="hidden" value="{{ csrf_token() }}">
     <div><label
-        for="organization_name">{{ gettext('Set the organization name used on the SecureDrop web interfaces:') }}</label>
+        for="organization_name">{{ gettext('Set the organization name used on the SecureDrop <abbr title="World Wide Web">web</abbr> interfaces:') }}</label>
```
Comment (cfm) [inline securedrop/journalist_templates/config.html:18]: I don't think "web" is worth annotating.  Users know what it means, and it's debatable whether SecureDrop's interfaces (as Tor onion services) are part of the *World Wide* Web.  :-)

File: securedrop/journalist_templates/config.html [hunk 7]
```diff
@@ -94,7 +94,7 @@ <h2 id="config-preventuploads">{{ gettext('Submission Preferences') }}</h2>
 <section aria-labelledby="config-testalert">
   <h2 id="config-testalert">{{ gettext('Alerts') }}</h2>
 
-  <p>{{ gettext('Send an encrypted email to verify if OSSEC alerts work correctly:') }}</p>
+  <p>{{ gettext('Send an encrypted email to verify if <abbr title="Open Source HIDS SECurity">OSSEC</abbr> alerts work correctly:') }}</p>
```
Comment (cfm) [inline securedrop/journalist_templates/config.html:97]: OSSEC is the name of an application, and we only ever refer to it this way.  I don't think it needs expanding.

File: securedrop/source_templates/footer.html [hunk 8]
```diff
@@ -3,7 +3,7 @@
     {{ gettext('Powered by') }} <b>SecureDrop {{ version }}</b>.
   </p>
   <p id="footer-advisory">
-    {{ gettext('Please note: Sharing sensitive information may put you at risk, even when using Tor and SecureDrop.') }}
+    {{ gettext('Please note: Sharing sensitive information may put you at risk, even when using <abbr title="The Onion Router">Tor</abbr> and SecureDrop.') }}
```
Comment (cfm) [inline securedrop/source_templates/footer.html:6]: I don't think this is worth expanding.  It's no longer the official initialism of the [Tor project or network](https://www.torproject.org/about/history/), and users will reach this page *via* Tor Browser already.

File: securedrop/source_templates/index.html [hunk 9]
```diff
@@ -22,7 +22,7 @@
   <div id="browser-security-level" class="warning" role="alert">
     <p>{{ gettext('Your Tor Browser\'s <a id="disable-js" href=""><b>Security Level</b></a> is too low. Use the <img src="{icon}" alt="&quot;Security Level&quot;"> button in your browser’s toolbar to change it.').format(icon=url_for("static", filename="i/font-awesome/white/guard.svg")) }}</p>
   </div>
-  {# Warning bubble to help TB users disable JavaScript with Tor Browser's Security Levels
+  {# Warning bubble to help <abbr title="Tor Browser">TB</abbr> users disable JavaScript with <abbr title="The Onion Router">Tor</abbr> Browser's Security Levels
```
Comment (cfm) [inline securedrop/source_templates/index.html:25]: This is a server-side [Jinja comment](https://jinja.palletsprojects.com/en/3.0.x/templates/#comments) for developers, so users will never see it, and HTML doesn't work here anyway.  :-)

File: securedrop/source_templates/logout.html [hunk 10]
```diff
@@ -10,7 +10,7 @@
 <section>
   <h2>{{ gettext('One more thing...') }}</h2>
   <p>
-    {{ gettext('Click the <img src={icon} alt="" width="16" height="16">&nbsp;<b>New Identity</b> button in your Tor Browser\'s toolbar. This will clear your Tor Browser activity data on this device.').format(icon=url_for('static', filename='i/torbroom.png'))  }}
+    {{ gettext('Click the <img src={icon} alt="" width="16" height="16">&nbsp;<b>New Identity</b> button in your <abbr title="The Onion Router">Tor</abbr> Browser\'s toolbar. This will clear your Tor Browser activity data on this device.').format(icon=url_for('static', filename='i/torbroom.png'))  }}
```
Comment (cfm) [inline securedrop/source_templates/logout.html:13]: See above.

File: securedrop/source_templates/tor2web-warning.html [hunk 11]
```diff
@@ -7,7 +7,7 @@ <h2>{{ gettext('Proxy Service Detected') }}</h2>
 <div class="center">
   {% include 'flashed.html' %}
 </div>
-<p>{{ gettext('You appear to be using a Tor proxy service to access SecureDrop. Proxy services do not protect your anonymity.') }}
+<p>{{ gettext('You appear to be using a <abbr title="The Onion Router">Tor</abbr> proxy service to access SecureDrop. Proxy services do not protect your anonymity.') }}
```
Comment (cfm) [inline securedrop/source_templates/tor2web-warning.html:10]: See above.

**Review by cfm — CHANGES_REQUESTED** [review]
@amnak1284, thanks for proposing these changes.  I've left some feedback on specific annotations.  In brief, some of these abbreviations benefit from annotation more than others in context, and we'd like to balance their benefit against their cost as changes requiring new translations.

**Comment by cfm** [comment]
@amnak1284, I'm going to close this pull request for now, but feel free to reopen it if you're able to make these changes!

File: securedrop/journalist_templates/account_new_two_factor.html (hunk 3 above)
Comment (amnak613) [reply securedrop/journalist_templates/account_new_two_factor.html:26]: fixed this change

File: securedrop/journalist_templates/admin.html (hunk 5 above)
Comment (amnak613) [reply securedrop/journalist_templates/admin.html:88]: fixed this

**Pull request closed without merging**